Disable the slave parts of twi

algernon opened the pull request:
The slave parts are unused, disable them. The code is kept, for easier merging, but guarded with an #if.

This saves almost 300 bytes of program space, and 71 bytes of SRAM. Pretty sizeable saving!

Thanks to @obra for the hint to look into this.

algernon committed:
Disable the slave parts of twi

The slave parts are unused, disable them. The code is kept, for easier merging,
but guarded with an #if.

This saves almost 300 bytes of program space, and 71 bytes of SRAM. Pretty
sizeable saving!

Thanks to @obra for the hint to look into this.

Signed-off-by: Gergely Nagy <algernon@madhouse-project.org>

obra commented:
It's a tiny dumb style thing, but having a define that's always used negated suggests to me that it should be inverted, say "ENABLE_TWI_SLAVE_MODE" or somesuch.

The concept is +100 :)

algernon committed:
Rename the slave-mode define

As it is always used negated, a better name would be `ENABLE_TWI_SLAVE_MODE`,
and used non-negated. Thanks to @obra for the suggestion!

Signed-off-by: Gergely Nagy <algernon@madhouse-project.org>

algernon commented:
Good idea, updated!

obra commented:
Merci!  Merged

swenson commented:
Nice!
